include/aspect/simulator.h

```diff
@@ -647,8 +648,10 @@ namespace aspect
        * @param advection_field Determines whether this variable should select
        * the temperature field or a compositional field.
        */
-      void get_artificial_viscosity (Vector<float> &viscosity_per_cell,
-                                     const AdvectionField &advection_field) const;
+      template <typename T>
+      void get_artificial_viscosity (Vector<T> &viscosity_per_cell,
+                                     const AdvectionField &advection_field,
+                                     const bool use_smoothing = false) const;
```

Comment:
I think you are calling this function with the default ```= false``` in some places in the code, is that correct? I think the postprocessor should also do the smoothing if it is used in the assembly.